staging/src/k8s.io/apiserver/pkg/registry/rest/create.go

```diff
@@ -105,6 +105,11 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx context.Context, obj runtime.
 		return errors.NewInternalError(fmt.Errorf("system metadata was not initialized"))
 	}
 
+	// ensure the name has been set
+	if len(objectMeta.GetName()) == 0 {
+		return errors.NewInternalError(fmt.Errorf("metadata.name was not set"))
```

Comment:
internal error here isn't right... a missing name should continue and be caught in validation and result in a Required validation error

Reply:
if you want to make sure generation happened, make this conditional on `len(objectMeta.GetGenerateName()) > 0`

Reply:
yeah,